crates/nu-parser/src/parse_keywords.rs

```diff
@@ -1005,7 +1005,27 @@ pub fn parse_alias(
             working_set.add_decl(Box::new(decl));
         }
 
-        if spans.len() < 4 {
+        // special case for `alias foo=bar`
+        if spans.len() == 2 && working_set.get_span_contents(spans[1]).contains(&b'=') {
+            // convert second argument to string.  Its a bit messy, could use
+            // refactoring.
+            let arg = working_set.get_span_contents(spans[1]).to_owned();
+            let arg = String::from_utf8_lossy(&arg);
+            // invariant: `arg` must contain the '=' byte from the if check
+            let index = arg.rfind('=').unwrap();
+            let (name, initial_value) = arg.split_at(index);
+            // remove first '=' from initial_value
+            // invariant: '=' must be in `initial_value`, because of `split_at`
+            // definition
+            let initial_value = &initial_value[1..];
```

Comment:
This can panic if `initial_value` is empty (e.g., `alias foo=`). It's best to avoid accesses with `[]` as much as possible.

Reply:
Actually, the first char should be `=`, so it shouldn't panic, but it's still better to remove the `[]` because it is error-prone. If the code gets moved or reused somewhere, it can cause panics.

Reply:
`foo=` gets separated into `foo` and `=`.  In general, `initial_value` must always start with `=`.  There should be a better way of parsing the argument, but I can't think of any.